api/src/main/java/io/minio/MinioClient.java

```diff
@@ -1049,6 +1049,17 @@ public void writeTo(BufferedSink sink) throws IOException {
     requestBuilder.method(method.toString(), requestBody);
     return requestBuilder.build();
   }
+  
+  /**
+   * Checks whether port should be omitted in Host header
+   * 
+   * HTTP Spec (rfc2616) defines that port should be omitted in Host header
+   * when port and service matches (i.e HTTP -> 80, HTTPS -> 443)
+   * @param url Url object
+   */
+  private boolean shouldOmitPortInHostHeader(HttpUrl url) {
+	  return (url.scheme().equals("http") && url.port() == 80) || (url.scheme().equals("https") && url.port() == 443);
```

Comment:
Wondering someone is using non-standard port for TLS and non-TLS i.e. for TLS port 80 is used and for non-TLS port 443 is used.

Is this your problem or just making spec compatible?

Reply:
I also have this problem but the change is making the code spec compatible.

Reply:
Got it.